httpx/client.py

```diff
@@ -109,6 +111,15 @@ def merge_cookies(
             return merged_cookies
         return cookies
 
+    def merge_headers(
+        self, headers: HeaderTypes = None
+    ) -> typing.Optional[HeaderTypes]:
+        if headers and self.headers:
+            merged_headers = Headers(self.headers)
+            merged_headers.update(headers)
+            return merged_headers
+        return headers or self.headers
```

Comment:
The reason you're getting the `NoneType is not iterable error` is because `Headers` does not implement the `update` method, [unlike `Cookies`](https://github.com/encode/httpx/blob/master/httpx/models.py#L1110-L1113), and falls back to the `MutableDict` implementation which does not handle the `None` case. Maybe it's worth implement it if only for consistency?

Reply:
Sounds good, I've implemented `update` to mimic Cookies now, and it appears to work just the same. :+1: